### PR TITLE
Responsive UI

### PR DESCRIPTION
**Pull request by MadalinaUngureanuG**

### CODE REVIEW

File: components/Weathercode.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+interface ComponentProps {
+    weathercode: number;
+    isDayOrNight: number;
+  }
+
+export default function Weathercode(props:ComponentProps) {
+    const {weathercode, isDayOrNight} = props;
+
+    return <>
```
Comment (madalina-gheorghe) [inline components/Weathercode.tsx:11]: Poate aici daca poti sa tratezi putin mai elegant tot ce e sub acest return, e putin greu de urmarit, mai ales de exemplu daca ar trebui sa lucreze si altcineva in zona asta

Comment (madalina-gheorghe) [reply components/Weathercode.tsx:11]: De exemplu, daca poti sa faci o functie care sa primeasca niste parametri si in functie de ei sa returneze ce ai tu nevoie aici

File: src/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	async function fetchWeatherData() {
+		const res = await fetch(`api/weather?latitude=${latitude}&longitude=${longitude}`);
+		let weather: Weather = await res.json();
+		setWeather({ ...weather });
+	}
+
+	function getCurrentPosition() {
+		if (navigator.geolocation) {
+			let position = navigator.geolocation.getCurrentPosition(position => {
+				setCity("Your current location");
```
Comment (madalina-gheorghe) [inline src/pages/index.tsx:40]: Aici vezi daca reusesti cumva sa gasesti si numele locatiei ca userul sa fie sigur ca ii ia locatia buna de exemplu

**Review by madalina-gheorghe — APPROVED** [review]

File: components/Weathercode.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      {(weathercode === 71 || weathercode === 73 || weathercode === 75) && <Card.Img src="icons/snow.svg" alt="snow image"/>} 
+      {weathercode === 77 && <Card.Img src="icons/hail.svg" alt="hail image"/>}
+      {weathercode === 80 && <Card.Img src={"icons/partly-cloudy-"+(isDayOrNight === 1 ? "day":"night")+"-rain.svg"} alt="partly cloudy day or night rain image"/>}
+      {(weathercode === 81 || weathercode === 82) && <Card.Img src="icons/rain.svg" alt="rain image"/>}
+      {weathercode === 85 && <Card.Img src={"icons/partly-cloudy-"+(isDayOrNight === 1 ? "day":"night")+"-snow.svg"} alt="partly cloudy day or night snow image"/>}
+      {weathercode === 86 && <Card.Img src="icons/snow.svg" alt="snow image"/>}
+      {weathercode === 95 && <Card.Img src={"icons/thunderstorms-"+(isDayOrNight === 1 ? "day":"night")+".svg"} alt="thunderstorms day or night drizzle image"/>}
+      {(weathercode === 96 || weathercode === 99) && <Card.Img src={"icons/thunderstorms-"+(isDayOrNight === 1 ? "day":"night")+"-snow.svg"} alt="thunderstorms day or night snow image"/>}
+    </>
+};
```
Comment (danielursu) [inline components/Weathercode.tsx:33]: aici ai putea refacoriza prin inlocuirea operatorilor ternari cu un switch


export default function Weathercode({ weathercode, isDayOrNight }: ComponentProps) {
  const renderImage = () => {
    const dayOrNight = isDayOrNight === 1 ? "day" : "night";

    switch (weathercode) {
      case 0:
      case 1:
        return `icons/clear-${dayOrNight}.svg`;
      case 2:
        return "icons/cloudy.svg";
      case 3:
        return `icons/overcast-${dayOrNight}.svg`;
      case 45:
        return `icons/fog-${dayOrNight}.svg`;
      case 48:
        return "icons/fog.svg";
      case 51:
        return `icons/partly-cloudy-${dayOrNight}-drizzle.svg`;
      case 53:
      case 55:
        return "icons/drizzle.svg";
      case 56:
      case 57:
        return "icons/hail.svg";
      case 61:
        return `icons/partly-cloudy-${dayOrNight}-rain.svg`;
      case 63:
      case 65:
        return "icons/rain.svg";
      case 66:
        return `icons/partly-cloudy-${dayOrNight}-sleet.svg`;
      case 67:
        return "icons/sleet.svg";
      case 71:
      case 73:
      case 75:
        return "icons/snow.svg";
      case 77:
        return "icons/hail.svg";
      case 80:
        return `icons/partly-cloudy-${dayOrNight}-rain.svg`;
      case 81:
      case 82:
        return "icons/rain.svg";
      case 85:
        return `icons/partly-cloudy-${dayOrNight}-snow.svg`;
      case 86:
        return "icons/snow.svg";
      case 95:
        return `icons/thunderstorms-${dayOrNight}.svg`;
      case 96:
      case 99:
        return `icons/thunderstorms-${dayOrNight}-snow.svg`;
      default:
        return "";
    }
  };

  return <>{renderImage() && <Card.Img src={renderImage()} alt={`${dayOrNight} weather image`} />}</>;
}
`

File: src/pages/api/cities.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+const getUrl = (name: string = '') => {
+	let url = 'https://api.api-ninjas.com/v1/city';
+	url += '?name=' + name;
+	return url;
+};
```
Comment (danielursu) [inline src/pages/api/cities.tsx:5]:
```suggestion
const BASE_URL = 'https://api.api-ninjas.com/v1/city';

function getUrl(name = '') {
  return `${BASE_URL}?name=${encodeURIComponent(name)}`;
}
```

File: src/pages/index.tsx [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+			const dayOfWeekName = formatter.format(new Date(weather.current_weather.time));
+			return dayOfWeekName;
+		}
+		return "";
+	}
+
+	function getVisibilityInKm(): number {
+		if (weather.hourly.visibility && weather.hourly.visibility.length > 0) {
+			const visibilityInKm: number = weather.hourly.visibility[0] / 1000;
+			return visibilityInKm;
```
Comment (danielursu) [inline src/pages/index.tsx:105]:
```suggestion
			return weather.hourly.visibility[0] / 1000;
```

**Review by danielursu — APPROVED** [review]

**Pull request merged**